clog/uwsgi_plugin.py

```diff
@@ -0,0 +1,58 @@
+import uwsgi
+import uwsgidecorators
+import logging
+import clog
+import clog.global_state
+import clog.handlers
+
+try:
+    import cPickle as pickle
+except ImportError:
+    import pickle
+
+
+def _mule_msg(*args, **kwargs):
+    mule = kwargs.pop('mule', 1)
+    data = pickle.dumps({
+        'service': 'uwsgi_mulefunc',
+        'func': 'log_line',
+        'args': args,
+        'kwargs': kwargs,
+    })
+    uwsgi.mule_msg(data, mule)
+
+
+class UwsgiHandler(logging.Handler):
+
+    def __init__(self, stream, mule=1):
+        logging.Handler.__init__(self)
+        self.stream = stream
+        self.mule = mule
+
+    def emit(self, record):
+        try:
+            msg = self.format(record)
+            _mule_msg(self.stream, msg, mule=self.mule)
+        except:
+            self.handleError(record)
+
+
+# TODO: Patch upstream uwsgi to return success on the pipe write so
+# we can optionally default to sync log_line
+def uwsgi_log_line(stream, line, mule=1):
+    _mule_msg(stream, line, mule=mule)
+
+
+def uwsgi_patch_global_state():
+    map(
+        lambda x: setattr(x, 'log_line', uwsgi_log_line),
+        (clog, clog.global_state)
+    )
+
+
+# Couple setup tasks at import:
+# 1. Register 'log_line' in all mules for dispatch handling on the mule side
+# 2. Insert 'UwsgiHandler' into the clog.handlers module for seamless usage
+
+uwsgidecorators.mule_functions['log_line'] = clog.global_state.log_line
+setattr(clog.handlers, 'UwsgiHandler', UwsgiHandler)
```

Comment:
In this case the entire module is conditionally included based on the present of uwsgi environment. In this case we want to modify the uwsgidecorators module at import time to guarantee the reference to the log_line function at post-fork, otherwise mule dispatches may fail. This could be ensured by encouraging users to initialize at app creation, but here it's a certainty and without any cost that I can see.

As for the monkey-patching of the UwsgiHandler line, sure we could add a:
```python
try:
    from clog.uwsgi_plugin import UwsgiHandler
except ImportError:
    pass
```
But the current approach keeps all the uwsgi related code inside this module and the \_\_init\_\_ file. I'm fine with either.